openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -780,6 +782,33 @@ def test_access_masquerade_as_user_with_forum_role(self, role_name):
             request_factory=self.factory,
         )
 
+    @ddt.data(True, False)
+    def test_discount_display(self, has_discount):
+        verified_mode = CourseMode.objects.get(course_id=self.course.id, mode_slug=CourseMode.VERIFIED)
+        verified_mode.min_price = 100
+        verified_mode.save()
+
+        with DISCOUNT_APPLICABILITY_FLAG.override(has_discount):
```

Comment:
See comment above about testing this separate from the discount applicability functions